pynamodb/connection/transactions.py

```diff
@@ -0,0 +1,134 @@
+        to ensure that we don't perform multiple operations on the same entry within the same transaction
+
+        :param model_cls:
+        :param hash_key:
+        :param range_key:
+        :return:
+        """
+        key = (model_cls, hash_key, range_key)
+        if key in self._hashed_models:
+            raise ValueError("Can't perform operation on the same entry multiple times in one transaction")
+        self._hashed_models.add(key)
+
+    @staticmethod
+    def _format_request_parameters(valid_parameters, operation_kwargs):
+        if RETURN_VALUES in operation_kwargs.keys():
+            operation_kwargs[RETURN_VALUES_ON_CONDITION_FAILURE] = operation_kwargs.pop(RETURN_VALUES)
+        return {
+            key: value for key, value in operation_kwargs.items() if key in valid_parameters
+        }
+
+
+class TransactGet(Transaction):
+
+    def __init__(self, *args, **kwargs):
+        super(TransactGet, self).__init__(*args, **kwargs)
+        self._get_items = []
+        self._futures = []
+
+    def get(self, model_cls, hash_key, range_key=None):
+        """
+        Adds the operation arguments for an item to list of models to get
+        returns a _ModelFuture object as a placeholder
+
+        :param model_cls:
+        :param hash_key:
+        :param range_key:
+        :return:
+        """
+        self._hash_model(model_cls, hash_key, range_key)
+        operation_kwargs = model_cls.get_operation_kwargs_for_get_item(hash_key, range_key=range_key)
+        get_item = self._format_request_parameters(TRANSACTION_GET_REQUEST_PARAMETERS, operation_kwargs)
+        model_future = _ModelFuture(model_cls)
+        self._futures.append(model_future)
+        self._get_items.append(get_item)
+        return model_future
+
+    def _update_futures(self):
+        for model, data in zip(self._futures, self._results):
+            model.update_with_raw_data(data[ITEM])
+
+    def _commit(self):
+        response = self._connection.transact_get_items(
+            get_items=self._get_items,
+            return_consumed_capacity=self._return_consumed_capacity
+        )
+        self._results = response[RESPONSES]
+        self._update_futures()
+
+
+class TransactWrite(Transaction):
+
+    def __init__(self, client_request_token=None, return_item_collection_metrics=None, **kwargs):
+        super(TransactWrite, self).__init__(**kwargs)
+        self._client_request_token = client_request_token
+        self._return_item_collection_metrics = return_item_collection_metrics
+        self._condition_check_items = []
+        self._delete_items = []
+        self._put_items = []
+        self._update_items = []
+
+    def add_condition_check_item(self, model_cls, hash_key, range_key, operation_kwargs):
+        condition_item = self._format_request_parameters(TRANSACTION_CONDITION_CHECK_REQUEST_PARAMETERS, operation_kwargs)
+        self._hash_model(model_cls, hash_key, range_key)
```

Comment:
Would it not be valid to have multiple condition checks on the same item? I think we only care that it's not one of the items modified in the transaction

Reply:
I could be wrong, but don't _think_ so? (Since you can add conditions to all of the write operations on their own, and the whole transaction fails if any of them fail)

But, at the same time, maybe it doesn't matter for us to validate that on this end, and we can just let people try/let the request failure error tell them what to fix

Reply:
removing!